arangod/Replication2/AgencyMethods.cpp

```diff
@@ -0,0 +1,150 @@
+      .cond(prevTerm.has_value(),
+            [&](auto&& precs) {
+              return std::move(precs).isEqual(path->currentTerm()->term()->str(),
+                                              prevTerm->value);
+            })
+      .end();
+}
+
+auto methods::updateTermSpecification(DatabaseID const& database, LogId id,
+                                      LogPlanTermSpecification const& spec,
+                                      std::optional<LogTerm> prevTerm)
+    -> futures::Future<Result> {
+  VPackBufferUInt8 trx;
+  {
+    VPackBuilder builder(trx);
+    updateTermSpecificationTrx(arangodb::agency::envelope::into_builder(builder),
+                               database, id, spec, prevTerm)
+        .done();
+  }
+
+  return sendAgencyWriteTransaction(std::move(trx));
+}
+
+auto methods::deleteReplicatedLogTrx(arangodb::agency::envelope envelope,
+                                     DatabaseID const& database, LogId id)
+    -> arangodb::agency::envelope {
+  auto path = paths::plan()->replicatedLogs()->database(database)->log(id)->str();
+
+  return envelope.write()
+      .remove(path)
+      .inc(paths::plan()->version()->str())
+      .precs()
+      .isNotEmpty(path)
+      .end();
+}
+
+auto methods::deleteReplicatedLog(DatabaseID const& database, LogId id)
+    -> futures::Future<Result> {
+  VPackBufferUInt8 trx;
+  {
+    VPackBuilder builder(trx);
+    deleteReplicatedLogTrx(arangodb::agency::envelope::into_builder(builder), database, id)
+        .done();
+  }
+
+  return sendAgencyWriteTransaction(std::move(trx));
+}
+
+auto methods::createReplicatedLogTrx(arangodb::agency::envelope envelope,
+                                     DatabaseID const& database,
+                                     LogPlanSpecification const& spec)
+    -> arangodb::agency::envelope {
+  auto path = paths::plan()->replicatedLogs()->database(database)->log(spec.id)->str();
+
+  return envelope.write()
+      .emplace_object(path,
+                      [&](VPackBuilder& builder) { spec.toVelocyPack(builder); })
+      .inc(paths::plan()->version()->str())
+      .precs()
+      .isEmpty(path)
+      .end();
+}
+
+auto methods::createReplicatedLog(DatabaseID const& database, LogPlanSpecification const& spec)
+    -> futures::Future<Result> {
+  VPackBufferUInt8 trx;
+  {
+    VPackBuilder builder(trx);
+    createReplicatedLogTrx(arangodb::agency::envelope::into_builder(builder), database, spec)
+        .done();
+  }
+
+  return sendAgencyWriteTransaction(std::move(trx));
+}
+
+auto methods::updateElectionResult(arangodb::agency::envelope envelope,
+                                   const DatabaseID& database, LogId id,
+                                   const LogCurrentSupervisionElection& result)
+    -> arangodb::agency::envelope {
+  auto path = paths::current()->replicatedLogs()->database(database)->log(to_string(id))->str();
+  return envelope.write()
+      .emplace_object(path + "/supervision/election",
```

Comment:
I don't question this works, but everywhere else around here paths are built using the AgencyPaths class and not via direct string concatenation. Is this an oversight here or intentional?

Reply:
This is probably an oversight. The agency paths work flow is a bit tedious when adding new paths to it. Sometimes you just want to get things done.

Maybe we should create a DSL that specifies the agency layout and creates those paths for us.

We will fix this to paths.